### ISSUE TITLE
Improve client logging

### ISSUE THREAD

**Issue by randomaccess3** [Velocidex/velociraptor#2856]
I'd like to start running the clients with the client logging enabled so that this can help troubleshoot any issues that are being seen. This can be enabled afterwards however you may not easily be able to retroactively create the issue that you're seen.
Proposed solution!

* Logging created with a max size of X, rotate older logs as required.
* Logging on disk is encrypted
* Logging can be collected using Velo and decrypted from the console

Will this have significant adverse issues to performance as the agent will be doing more and therefore consuming more resources?

(It would be super cool however if there was a troubleshooting section in the client section where there's a button for "enable logging", "collect logging", "restart agent", "killkillkill"

**Comment by mgreen27** [comment]
Just adding to this an issue I see running yara a lot.

I can add a malformed yara rule into a hunt and the hunt runs successfully with zero rows. 
Its not until you review the flow logs on the client you can see there was an issue with the yara rule.
We can probably resolve my issue with making a yara error an exception.

**Comment by scudette** [comment]
This is probably a separate issue. We can add parameter validation on artifact creation in the GUI to refuse to accept a hunt with invalid Yara rules in the first place

**Comment by scudette** [comment]
Fixed by #3063
